test/functional/test_framework/authproxy.py

```diff
@@ -160,6 +160,11 @@ def _get_response(self):
             raise JSONRPCException({
                 'code': -342, 'message': 'missing HTTP response from server'})
 
+        if http_response.status == HTTPStatus.NO_CONTENT:
+            if len(http_response.read()) != 0:
+                raise JSONRPCException({'code': -342, 'message': 'Content received with NO CONTENT status code'})
+            return None, http_response.status
+
```

Comment:
In commit "test: handle 204 NO_CONTENT correctly in authproxy" (174a995edf594dccf745d2593c281d3e4a1ff7f7)

It would be good to have a comment that explains why this block of code is needed, because it seems like an impossible case to hit through normal usage of authproxy. Maybe: "# Check for no-content HTTP status code, which can be returned when an RPC client requests a JSON-RPC 2.0 "notification" with no response. Currently this is only possible if clients call the _request() method directly to send a raw request."